### DISCUSSION TITLE
Execution failed

### DISCUSSION THREAD

**Discussion by LaDa26** [nrc-cnrc/EGSnrc#1203] [Questions]
Hi!

I am new to EGSnrc and currently going through the Getting Started documentation (the myapp example).
Compiling seems to work, however when executing the myapp application it seems to fail as it says ---Execution failed---.
I also followed the suggestion of trying compiling egs_app (as suggested in https://github.com/nrc-cnrc/EGSnrc/discussions/820), but I get the same error when executing. I am working on Windows.
Any idea how I can fix this problem? 

Thanks in advance for the help

![afbeelding](https://github.com/user-attachments/assets/0612b24c-a3a4-4091-b59a-527d6c2265f3)

![afbeelding](https://github.com/user-attachments/assets/ca844e03-e0c6-4887-a977-548508dd28b5)

**Comment by rtownson** [comment]
There might just be a typo in your input file. You could post it here. You could also try running on the command line to see if you get any additional output:
```
myapp -i slab
```

**Reply by LaDa26** [reply]
Running on the command line does not give any additional output. 
The input file is as follows

```
################################
#
# Simple slab simulation
#
################################


################################
### RUN CONTROL
################################
:start run control:
    ncase = 1e3 # The number of histories to simulate
:stop run control:


################################
### GEOMETRY
################################
:start geometry definition: # Only 1 geometry definition block allowed

    ### Define a plate of tantalum
    :start geometry: # Many geometry blocks can be defined
        name = slab # This name can be anything you like
        library = egs_ndgeometry
        type = EGS_XYZGeometry
        x-planes = -5, 5 # in cm
        y-planes = -5, 5 # in cm
        z-planes = -10, 0, 0.1, 10 # in cm
        :start media input:
            media = vacuum tantalum # Indexed as 0 1
            set medium = 1 1 # Set region 1 to medium 1 (tantalum)
                            # Other regions default to medium 0 (vacuum)
        :stop media input:
    :stop geometry:

    ### Use the geometry with this name for the simulation
    simulation geometry = slab

:stop geometry definition:


################################
### MEDIA
################################
:start media definition: # Only 1 media definition block allowed

    # Defining media in the input file like this is called "pegsless" mode

    ### Cutoff energies, in MeV
    ae = 0.521 # lowest energy for electrons (kinetic+0.511)
    ap = 0.01 # lowest energy for photons (kinetic)
    ue = 50.511 # maximum energy for electrons (kinetic+0.511)
    up = 50 # maximum energy for photons (kinetic)

    ### Tantalum
    :start tantalum: # this name can be anything
        density correction file = tantalum # name the file (no ext.)
    :stop tantalum:

    ### Lead
    :start lead:
        density correction file = lead
    :stop lead:

    ### Water
    :start water:
        density correction file = water_liquid
    :stop water:
:stop media definition


################################
### SOURCE
################################
:start source definition: # Only 1 source definition block allowed

    ### Pencil beam
    :start source:
        name = pencil_beam # This name can be anything you like
        library = egs_parallel_beam
        charge = -1
        direction = 0 0 1
        :start spectrum:
            type = monoenergetic
            energy = 20 # in MeV
        :stop spectrum:
        :start shape:
            type = point
            position = 0 0 -10 # in cm
        :stop shape:
    :stop source:

    ### Use the source by this name for the simulation
    simulation source = pencil_beam

:stop source definition:


################################
### VIEWER CONTROL
################################
:start view control:
    # Here we are just assigning some colors for nice
    # viewing in the egs_view application
    set color = lead 120 120 200 200
    set color = tantalum 120 255 120 255
    set color = water 0 220 255 200
:stop view control:


################################
### AUSGAB OBJECTS
################################
:start ausgab object definition: # Only 1 ausgab definition block allowed
    ### Particle tracks
    :start ausgab object:
        name = tracks
        library = egs_track_scoring
    :stop ausgab object:
:stop ausgab object definition:

```

**Reply by rtownson** [reply]
The input file works for me. Is it really just entirely blank output when you run myapp on the command line? It doesn't say 'command not found' or anything? 

Try copying the input file into the `egs_home/egs_app` directory and running it there. You might need to compile egs_app first.

**Reply by rtownson** [reply]
Make sure that the input file has the .egsinp extension. Not .egsinp.txt or anything else.

**Reply by LaDa26** [reply]
I verified that the file is saved as an .egsinp extension. I get the same problem when running egs_app with this input_file. Running with egs_gui still gives "execution failed" and running in command line doesn't give any output at all.

![afbeelding](https://github.com/user-attachments/assets/93232131-0fdd-43ca-b67b-32917af80d0a)

**Comment by rtownson** [comment]
Pretty weird! Could try a `mingw32-make clean` then `mingw-make` and watch for errors or warnings, but seems unlikely that will help. Also, double check that `slab.egsinp` exists in `C:\EGSnrc\egs_home\egs_app\`, it has to be in that directory and not in a sub directory.

**Reply by LaDa26** [reply]
Somehow it still doesn't work, unfortunately. Slab.egsinp is in the correct folder as I am able to visualize the geometry. However, the simulation itself fails. Could it be that something went wrong during installation? 
[config_win3264.log](https://github.com/user-attachments/files/17699965/config_win3264.log)
